br/pkg/backup/client.go

```diff
@@ -1000,6 +1000,38 @@ func WriteBackupDDLJobs(metaWriter *metautil.MetaWriter, g glue.Glue, store kv.S
 	if err != nil {
 		return errors.Trace(err)
 	}
+
+	handleJobsFn := func(jobs []*model.Job) (bool, error) {
+		for _, job := range jobs {
+			if skipUnsupportedDDLJob(job) {
+				continue
+			}
+
+			if (job.State == model.JobStateDone || job.State == model.JobStateSynced) &&
+				(job.BinlogInfo != nil && job.BinlogInfo.SchemaVersion > lastSchemaVersion && job.BinlogInfo.SchemaVersion <= backupSchemaVersion) {
+				if job.BinlogInfo.DBInfo != nil {
+					// ignore all placement policy info during incremental backup for now.
+					job.BinlogInfo.DBInfo.PlacementPolicyRef = nil
+				}
+				if job.BinlogInfo.TableInfo != nil {
+					// ignore all placement policy info during incremental backup for now.
+					job.BinlogInfo.TableInfo.ClearPlacement()
+				}
+				jobBytes, err := json.Marshal(job)
+				if err != nil {
+					return true, errors.Trace(err)
+				}
+				err = metaWriter.Send(jobBytes, metautil.AppendDDL)
+				if err != nil {
+					return true, errors.Trace(err)
+				}
+			}
+			// early exits to stop unnecessary scan
+			return true, nil
```

Comment:
The early exit condition is wrong. The `if` condition on line 1010 can fail in cases like `state != Done` or `SchemVersion > backupSchemaVersion` which you need to continue scanning.

The only good stop condition is `SchemaVersion <= lastSchemaVersion`.